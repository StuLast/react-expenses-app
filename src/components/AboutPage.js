import React from "react";
import { Link } from "react-router-dom";

export const AboutPage = () => <div></div>;

export default AboutPage;
