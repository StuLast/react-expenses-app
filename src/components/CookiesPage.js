import React from "react";
import { Link } from "react-router-dom";

export const CookiesPage = () => (
  <div>
    <h1>About Page</h1>
  </div>
);

export default CookiesPage;
