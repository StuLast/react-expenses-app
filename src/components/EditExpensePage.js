import React from "react";

const EditExpensePage = (props) => {
  console.log(props.match.params);
  return (
    <div>
      <p>This is from my Edit Expense Component</p>
      <p>Editing the expense with ID of {props.match.params.id}.</p>
    </div>
  );
};

export default EditExpensePage;
