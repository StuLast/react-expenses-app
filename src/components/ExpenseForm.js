import React from 'react';;

class ExpenseForm extends React.Component {

  state = {
    description: "",
    note: "", 
    amount: 0
  };

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = (e) => {
    const amount = e.target.value;
    if(amount.match(/^\d*(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }))
    }
  };

  onSubmit = (e) => {
    e.preventDefault();
  };

  render () {
    return (
    <div>
      <form>
        <input 
          type="text"
          placeholder="Description"
          autofocus
          value = {this.state.description}
          onChange = {this.onDescriptionChange}
        />
        <input
          type="text"
          placeholder="Amount"
          value={this.state.amount}
          onChange={this.onAmountChange}
        />
        <textarea
          placeholder="Add a note for your expense (optional)"
          value = {this.state.note}
          onChange = {this.onNoteChange}
        >
        </textarea>
        <button onClick={this.onSubmit}>Add Expense</button>
      </form>
    </div>
    )
  };
}

export default ExpenseForm;
