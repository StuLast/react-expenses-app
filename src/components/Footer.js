import React from "react";
import { Link } from "react-router-dom";

export const Footer = () => (
  <footer className="footer">
    <div className="footer__top">
      <div className="footer__column">
        <h3>Links:</h3>
        <Link to="/about">About this demo</Link>
      </div>
      <div className="footer__column"></div>
      <div className="footer__column">
        <h3>Privacy:</h3>
        <Link to="/privacy">Privacy</Link>
        <Link to="/cookies">Cookies</Link>
      </div>
    </div>
    <div className="footer__bottom">
      <div>
        Powered by:
        <Link to="https://spycedconcepts.uk" target="_blank">
          Spyced Concepts
        </Link>
        . Programmed by:
        <Link href="https://stulast.co.uk" target="_blank">
          Stu Last
        </Link>
      </div>
    </div>
  </footer>
);

export default Footer;
