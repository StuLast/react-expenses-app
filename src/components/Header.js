import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Expensologist</h1>
        </Link>
        <button className="button button--link" onClick={startLogout}>
          Log out
        </button>
      </div>
      <div className="header__data-warning">
        This app is only a demo app. Data and data structures may be deleted or
        changed without warning.
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
