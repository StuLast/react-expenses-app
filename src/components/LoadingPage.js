import React from "react";

export const LoadingPage = () => (
  <div className="loader">
    <img
      className="loader__image"
      src="/images/loader.gif"
      alt="loading screen animation"
    />
    <div>Loading...</div>
  </div>
);

export default LoadingPage;
