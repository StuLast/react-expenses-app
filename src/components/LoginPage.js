import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensologist</h1>
      <p>React.js Demo App - Expenses Manager</p>
      <button className="button" onClick={startLogin}>
        Log In with Google
      </button>
      <Link className="small-link" to="/about">
        about this site ...
      </Link>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

export { LoginPage };
