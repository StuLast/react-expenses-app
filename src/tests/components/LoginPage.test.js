import React from "react";
import { shallow } from "enzyme";

import { LoginPage } from "../../components/LoginPage";

let startLogin, wrapper;
beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(<LoginPage startLogin={startLogin} />);
});

describe("Render Login Page:", () => {
  it("should render the login page correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Login UI tests", () => {
  it("should call startLogin on button click", () => {
    wrapper.find("button").simulate("click");
    expect(startLogin).toHaveBeenCalled();
  });
});
