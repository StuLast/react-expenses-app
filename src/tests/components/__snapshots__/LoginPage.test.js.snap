// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Page: should render the login page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensologist
    </h1>
    <p>
      React.js Demo App - Expenses Manager
    </p>
    <button
      className="button"
      onClick={[MockFunction]}
    >
      Log In with Google
    </button>
    <Link
      className="small-link"
      to="/about"
    >
      about this site ...
    </Link>
  </div>
</div>
`;
